### PR TITLE
Expose the platform view mutator stack to custom compositors.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This allows custom compositors to affect scene builder modifications made
to the platform view.

Fixes https://github.com/flutter/flutter/issues/44211
Fixes b/143612326

### CODE REVIEW

File: shell/platform/embedder/embedder.h [hunk 1]
```diff
@@ -661,13 +679,55 @@ typedef struct {
   VoidCallback destruction_callback;
 } FlutterSoftwareBackingStore;
 
+typedef enum {
```
Comment (dnicoara) [inline shell/platform/embedder/embedder.h]: Could you keep enum value naming convention consistent? ie: kFlutterPlatformViewMutationType*

Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Yeah, my bad. Fixed.

File: shell/platform/embedder/embedder.h [hunk 2]
```diff
@@ -661,13 +679,55 @@ typedef struct {
...
+  /// Indicates that the Flutter application requested that an opacity be
+  /// applied to the platform view.
+  FlutterPlatformViewMutationTypeOpacity,
+  /// Indicates that the Flutter application requested that the platform view be
+  /// clipped using a rectangle.
+  FlutterPlatformViewMutationTypeClipRect,
+  /// Indicates that the Flutter application requested that the platform view be
+  /// clipped using a rounded rectangle rectangle.
```
[the remark this answers is not in the document]
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Fixed.

File: shell/platform/embedder/embedder.h [hunk 3]
```diff
@@ -661,13 +679,55 @@ typedef struct {
...
   /// The identifier of this platform view. This identifier is specified by the
   /// application when a platform view is added to the scene via the
   /// `SceneBuilder.addPlatformView` call.
   FlutterPlatformViewIdentifier identifier;
+  /// The number of mutations to be applied to the platform view by the embedder
+  /// before on-screen composition.
+  size_t mutations_count;
+  /// The mutations to be applied by this platform view before it is composited
+  /// on-screen. The Flutter application may transform the platform view but
+  /// these transformations cannot be affected by the Fluter compositor because
```
[the remark this answers is not in the document]
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Done.

File: shell/platform/embedder/embedder.h [hunk 4]
```diff
@@ -661,13 +679,55 @@ typedef struct {
...
+  /// it does not render platform views. Since the embedder is responsible for
+  /// composition of these view, it is also embedder responsibility to affect
```
[the remark this answers is not in the document]
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Done.

**Pull request merged**